### PR TITLE
[WIP] Autorouting

### PR DESCRIPTION
**Pull request by NickHeiner**
This is very preliminary and there's a ton wrong with it.

### CODE REVIEW

File: assets/app/view/game/autoroute.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module View
```
Comment (tobymao) [inline assets/app/view/game/autoroute.rb:1]: the calculation should be in the lib/engine, but only called in the views

File: assets/app/view/game/autoroute.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  module Game
+    class Autoroute
+      def self.calculate(game, corporationId)
```
Comment (tobymao) [inline assets/app/view/game/autoroute.rb:4]: always use snake_case

File: assets/app/view/game/autoroute.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        corp = game.corporation_by_id(corporationId)
+        trains = corp.trains
+
+        if trains.empty?
+          return []
```
Comment (tobymao) [inline assets/app/view/game/autoroute.rb:9]: in ruby, the last line is always return, so you don't need to add a return

File: assets/app/view/game/autoroute.rb (hunk 2 above)
Comment (NickHeiner) [reply assets/app/view/game/autoroute.rb:4]: Heh! Of course. I haven't run rubocop on this but obviously I will before submitting this for real.

File: assets/app/view/game/autoroute.rb (hunk 3 above)
Comment (NickHeiner) [reply assets/app/view/game/autoroute.rb:9]: I added `return` here because I'm going to add more after it.

**Pull request closed without merging**